crates/ra_assists/src/handlers/unwrap_block.rs

```diff
@@ -0,0 +1,435 @@
+use crate::{Assist, AssistCtx, AssistId};
+
+use ast::LoopBodyOwner;
+use ra_fmt::unwrap_trivial_block;
+use ra_syntax::{ast, AstNode};
+
+// Assist: unwrap_block
+//
+// This assist removes if...else, for, while and loop control statements to just keep the body.
+//
+// ```
+// fn foo() {
+//     if true {<|>
+//         println!("foo");
+//     }
+// }
+// ```
+// ->
+// ```
+// fn foo() {
+//     println!("foo");
+// }
+// ```
+pub(crate) fn unwrap_block(ctx: AssistCtx) -> Option<Assist> {
+    let res = if let Some(if_expr) = ctx.find_node_at_offset::<ast::IfExpr>() {
+        // if expression
+        let mut expr_to_unwrap: Option<ast::Expr> = None;
+        for block_expr in if_expr.blocks() {
+            if let Some(block) = block_expr.block() {
+                let cursor_in_range =
+                    block.l_curly_token()?.text_range().contains_range(ctx.frange.range);
+
+                if cursor_in_range {
+                    let exprto = unwrap_trivial_block(block_expr);
+                    expr_to_unwrap = Some(exprto);
+                    break;
+                }
+            }
+        }
+        let expr_to_unwrap = expr_to_unwrap?;
+        // Find if we are in a else if block
+        let ancestor = ctx
```

Comment:
`ancestors_with_macros` search with macros, which means it is possible to search node with different root. I think it is okay to not support macro at this PR. 

You could use `if_expr.syntax().ancestors()` directly here.